diff --git a/dist/changesets-assemble-release-plan.cjs.js b/dist/changesets-assemble-release-plan.cjs.js
index 4f7b5e5b37bb05874a5c1d8e583e29d4a9593ecf..1bf0b8aa3dfa75364cbf10f9d8cb81f24c04b5be 100644
--- a/dist/changesets-assemble-release-plan.cjs.js
+++ b/dist/changesets-assemble-release-plan.cjs.js
@@ -317,12 +317,23 @@ function getDependencyVersionRanges(dependentPkgJSON, dependencyRelease) {
     if (!versionRange) continue;
 
     if (versionRange.startsWith("workspace:")) {
+      // intentionally keep other workspace ranges untouched
+      // this has to be fixed but this should only be done when adding appropriate tests
+      let workspaceRange = versionRange.replace(/^workspace:/, "");
+      switch (workspaceRange) {
+        case "*":
+          // workspace:* actually means the current exact version, and not a wildcard similar to a reguler * range
+          workspaceRange = dependencyRelease.oldVersion;
+          break;
+        case "~":
+        case "^":
+          // Use ^oldVersion for workspace:^ or ~oldVersion for workspace:~.
+          // The version range might have changed in dependent package, but that should have its own changeset bumping that package.
+          workspaceRange += dependencyRelease.oldVersion;
+      }
       dependencyVersionRanges.push({
         depType: type,
-        versionRange: // intentionally keep other workspace ranges untouched
-        // this has to be fixed but this should only be done when adding appropriate tests
-        versionRange === "workspace:*" ? // workspace:* actually means the current exact version, and not a wildcard similar to a reguler * range
-        dependencyRelease.oldVersion : versionRange.replace(/^workspace:/, "")
+        versionRange: workspaceRange,
       });
     } else {
       dependencyVersionRanges.push({
